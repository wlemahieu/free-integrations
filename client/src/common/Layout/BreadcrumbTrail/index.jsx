import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Breadcrumb } from 'antd';
import getBreadcrumbs from 'utilities/getBreadcrumbs';

const BreadcrumbTrail = props => {
  const breadcrumbs = getBreadcrumbs(props.location);
  return (
    <Breadcrumb>
      {breadcrumbs.map(crumb => (
        <Breadcrumb.Item key={crumb.path}>
          <Link to={crumb.to} title={crumb.title}>
            {crumb.title}
          </Link>
        </Breadcrumb.Item>
      ))}
    </Breadcrumb>
  );
};

BreadcrumbTrail.propTypes = {
  location: PropTypes.object
};

BreadcrumbTrail.defaultProps = {
  location: {}
};

export default BreadcrumbTrail;
