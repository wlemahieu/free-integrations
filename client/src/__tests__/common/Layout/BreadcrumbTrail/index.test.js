import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render, getAllByTitle } from '@testing-library/react';
import BreadcrumbTrail from 'common/Layout/BreadcrumbTrail';

let wrapper;
beforeEach(() => {
  wrapper = render(
    <Router>
      <BreadcrumbTrail location={{ pathname: '/chat' }} />;
    </Router>
  );
});
afterEach(() => {
  wrapper = undefined;
});

describe('Layout Body', () => {
  it('should match snapshot', () => {
    const { container } = wrapper;
    expect(container).toMatchSnapshot();
  });
  it('should render "Home" crumb', async () => {
    const { container } = wrapper;
    const crumb = await getAllByTitle(container, 'Home');
    expect(crumb.length).toEqual(1);
  });
  it('should render "Chat" crumb', async () => {
    const { container } = wrapper;
    const crumb = await getAllByTitle(container, 'Chat');
    expect(crumb.length).toEqual(1);
  });
});
