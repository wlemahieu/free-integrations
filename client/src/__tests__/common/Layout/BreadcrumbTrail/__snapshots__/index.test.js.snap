// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Body should match snapshot 1`] = `
<div>
  <div
    class="ant-breadcrumb"
  >
    <span>
      <span
        class="ant-breadcrumb-link"
      >
        <a
          href="/home"
          title="Home"
        >
          Home
        </a>
      </span>
      <span
        class="ant-breadcrumb-separator"
      >
        /
      </span>
    </span>
    <span>
      <span
        class="ant-breadcrumb-link"
      >
        <a
          href="/chat"
          title="Chat"
        >
          Chat
        </a>
      </span>
      <span
        class="ant-breadcrumb-separator"
      >
        /
      </span>
    </span>
  </div>
  ;
</div>
`;
